tools/mason/MasonSearch.chpl

```diff
@@ -85,16 +91,21 @@ proc isHidden(name : string) : bool {
 proc findLatest(packageDir) {
   var ret = new VersionInfo(0, 0, 0);
   const suffix = ".toml";
-
+  var foundValidVersion = false;
   for fi in listdir(packageDir, files=true, dirs=false) {
-    assert(fi.endsWith(suffix));
-    const end = fi.length - suffix.length;
-    const ver = new VersionInfo(fi[1..end]);
-
-    if ver > ret then ret = ver;
+    if fi.endsWith(suffix) {
+      const end = fi.length - suffix.length;
+      const ver = new VersionInfo(fi[1..end]);
+      if ver > ret then ret = ver;
+      foundValidVersion = true;
+    }
+    else {
+      var warning_str = "file with bad/(not a .toml) extension encountered. skipping ";
```

Comment:
Stylistic nitpick: Chapel identifiers tend to be `camelCase` over `under_scored`, so I'd call this `warningMsg`, e.g.

Reply:
Suggested rewording, instead of:

```
file with bad/(not a .toml) extension encountered. skipping 0.5.0  in package /Users/anubhavujjawal/Desktop/work/gsoc/contrib/chapel/test/mason/search/badFileName/.mason/registry/Bricks/badFileName
```

How do you feel about:

```
File without '.toml' extension encountered - skipping <package> <version>
```

We can get package name from the last directory in `packageDir` with [`basename`](https://chapel-lang.org/docs/master/modules/standard/Path.html#Path.basename) :

```chpl
var packageName = basename(packageDir);
```

We may want to double check that `packageDir` does not have a trailing `/` though.